packages/core/src/request-builder/header-builder/authorization-header.ts

```diff
@@ -221,8 +222,9 @@ async function getAuthenticationRelatedHeaders(
 export async function buildAuthorizationHeaders(
   destination: Destination
 ): Promise<MapType<string>> {
+  const sanitizedDestination = sanitizeDestination(destination);
```

Comment:
I think this makes sense but could it change the behaviour of exisinting code. Assume the case a did not set the authType. In the past the value was not defaulted and nothing was done in the `getAuthenticationRelatedHeaders`. Now this is defaulted to basic in case there is a user and pwd. Which makes sense but is a difference. But you could consider it as a bug fix.

Reply:
The only design question is if you should use it there? Should the sanatize not be done a bit higher in the stack? Or if you do it for the defaulting of the authType only then make this more explicit.

Reply:
This is a very good point. The sanitize is done in our internal flow. I just did not think of it this way as I was blinded by the previous state a bit.
Therefore, if you use this with our clients it doesn't make a difference.
If you use it on your own, we should decide what we want the behavior to be. Should it be a prerequisite to call sanitize on your own or should we call it interally? If we call it internally I think it will be a bit more practical. On the other hand if we separate this, we have a stronger separation of concerns. We should probably have two different interfaces that make it clear whether we are expecting a destination or a sanitized destination, but rather for SDK 2.0 than now.
I am ok with both approaches, what do you think?

Reply:
Please add an issue with the 2.0 label on it so that we do not forget about this. For me also both ways are fine as long as the js doc tells the user if either a sanatized destination is to be used or the destination gets sanitized. Currently if I see it right we use `assoc` to add the properties in the sanitize process which is fine in principal, but only for primitive values. So for example the proxyConfiguration gets copied by reference which I would see as a source of a potential mean side effect in case something is adjusted on the sanatized object the initial destination is changed. When you have this in mind one should always have a strict function scope.

Reply:
I agree. I added a ticket. I am considering also removing the sanitization. Maybe @mr-flannery do you have an opinion on this?

Reply:
I very much agree that we need a "type answer" to the question of whether we expect a "sanitized destination" or not. Having two interfaces is not necessarily the ultimate answer in the general case, but should be sufficient here. But we need to find a way to keep the responsibility of transforming data into a useable state out of functions like this. I'm fine with this change here, since I also agree that it makes the whole thing usable in more scenarios.

Lemme go on a tangent here to maybe spark some discussion: Generally, what I think we need is a way to decouple the general schema of a thing from what we need from a specific instance of that thing in specific use case. 
So right now almost all the fields of the `Destination` interface are marked as optional since we cannot be sure which stuff is present together with which other stuff. So I guess the Java answer to that problem (and I think that's what the Java SDK actually did for v3) would be to add more concrete classes, e.g. `Destination` which is extended by `HttpDestination` which is extended by `BasicHttpDestination` which is extended by `BasicOnPremiseHttpDestination` (I'm making stuff up, but you get the idea). 
TypeScript kind of almost offers a solution to that problem. For example you could the `Pick` and `Partial` type:
```
Pick<Destination, 'url' | 'authType' | 'username' | 'password'> & Partial<Destination>
```
So this would give a type that requires url, authType, username and password from a destination and makes everything else optional.
However, if you require a lot of properties from a thing, this quickly becomes hard to read. Also you'd have to still name all such variants, since inline types are not properly picked up by typedoc. I think if you'd use this thing as an inline type, typedoc would simply render it as `object`, which is extremely unhelpful...

So yeah, guess there's still room for improvement in the world of typesystems.

Reply:
Ok, I created a ticket in the backlog, lets take a closer look at this for v2. Thanks for your opinions!